jujugui/static/gui/src/app/components/deployment-flow/direct-deploy/direct-deploy.js

```diff
@@ -50,64 +58,104 @@ class DeploymentDirectDeploy extends React.Component {
     });
   }
 
-  render() {
-    let diagram = null;
-    let titleAndDescription = null;
+  /**
+    Generate the entity link.
+
+    @returns {Object} The link markup or null.
+  */
+  _generateLink() {
     const ddEntityId = this.props.ddData.id;
+    let url;
+    try {
+      url = window.jujulib.URL.fromLegacyString(ddEntityId);
+    } catch(_) {
+      url = window.jujulib.URL.fromString(ddEntityId);
+    }
+    url = this.props.generatePath({
+      store: url.path()
+    });
+    if (!url) {
+      return null;
+    }
+    return (
+      <a className="link"
+        href={`${url}`}
+        target="_blank">
+        Learn more about this {this.state.isBundle ? 'bundle' : 'charm'}.
+      </a>);
+  }
+
+  /**
+    Generate the entity diagram or icon.
+
+    @returns {Object} The markup.
+  */
+  _generateImage() {
     if (this.state.isBundle) {
-      diagram = (
+      return (
         <juju.components.EntityContentDiagram
           getDiagramURL={this.props.getDiagramURL}
-          id={ddEntityId} />);
+          id={this.props.ddData.id} />);
+    } else {
+      const entity = this.state.entityModel.toEntity();
+      return (
+        <div className="deployment-direct-deploy__image-block">
+          <img alt={entity.displayName}
+            className="deployment-direct-deploy__image-block-icon"
+            src={entity.iconPath}
+            width="96" />
+        </div>);
     }
+  }
 
-    if (this.state.entityModel) {
+  render() {
+    let content = null;
+    if (this.state.loading) {
+      content = (<juju.components.Spinner />);
+    } else if (!this.state.entityModel) {
+      content = (
+        <div>
+          This {this.state.isBundle ? 'bundle' : 'charm'} could not be found.
```

Comment:
Now what? Maybe a comment for them to go back to the canvas (with a link) so they can search for themselves? Or maybe to the store?